syntax = "proto3";
package milvus.proto.etcd;
option go_package="github.com/zilliztech/milvus-distributed/internal/proto/etcdpb";

import "index_builder.proto";
import "common.proto";
import "schema.proto";

message TenantMeta {
  int64 ID = 1;
  int64 num_query_nodes = 2;
  repeated string insert_channelIDs = 3;
  string query_channelID = 4;
}


message ProxyMeta {
  int64 ID = 1;
  common.Address address = 2;
  repeated string result_channelIDs = 3;
}


message CollectionMeta {
  int64 ID=1;
  schema.CollectionSchema schema=2;
  uint64 create_time=3;
  repeated int64 segmentIDs=4;
  repeated string partition_tags=5;
}

message FieldBinlogFiles {
  int64 fieldID = 1;
  repeated string binlog_files = 2;
}

message SegmentMeta {
  int64 segmentID=1;
  int64 collectionID =2;
  string partition_tag=3;
  int32 channel_start=4;
  int32 channel_end=5;
  uint64 open_time=6;
  uint64 close_time=7;
  int64 num_rows=8;
  int64 mem_size=9;
  repeated FieldBinlogFiles binlog_file_paths = 10;
}

message FieldIndexMeta {
  int64 segmentID = 1;
  int64 fieldID = 2;
  int64 indexID = 3;
  repeated common.KeyValuePair index_params = 4;
  service.IndexStatus status = 5;
  repeated string index_file_paths = 6;
}